src/boot_semver/core.clj

```diff
@@ -49,9 +50,13 @@
 
 (defn- update-version [vermap upmap]
   (let [res #(-> % symbol resolve)]
-    (merge-with (fn [uv vv] (if (res uv)
-                              ((res uv) (or vv 0))
-                              (util/exit-error (util/fail "Unable to resolve symbol: %s \n" uv)))) upmap vermap)))
+    (merge-with
+      (fn [uv vv]
+        (if (res uv)
+          ((res uv) (if (string? vv)
+                      (-> vv (cstr/strip-prefix "+") (cstr/strip-prefix "-") read-string)
```

Comment:
@flyboarder just a suggestion, why not using `clojure.string` instead without including a dependency here?
It would have been just a simple `clojure.string/replace` I think and you would've been able to do it in one go...